File: TypeScript/sort/bubbleSort.ts
function bubbleSort(list: number[]): number[] {
  const len = list.length
  for (let i = 0; i < len; i++) {
    for (let j = 0; j < len - 1; j++) {
      if (list[j] > list[j + 1]) {
        const temp = list[j]
        list[j] = list[j + 1]
        list[j + 1] = temp
      }
    }

  }
  return list
}


console.log(bubbleSort([1, 3, 2, 4, 5, 0]))

File: rust/src/sort/bubble_sort.rs
fn bubble_sort<T: Ord>(list: &mut [T]) {
    if list.is_empty() {
        return;
    }
    let len = list.len();
    for _ in 0..len {
        for j in 0..len - 1 {
            if list[j] > list[j + 1] {
                list.swap(j, j + 1)
            }
        }
    }
}

#[cfg(test)]
mod test {

    use super::*;
    #[test]
    fn bubble_sort_test() {
        let mut list = [1, 3, 2, 5, 4];
        let expected = [1, 2, 3, 4, 5];
        bubble_sort(&mut list);
        assert_eq!(list, expected)
    }
}
